File: Fortran/FizzBuzz.f90
program FizzBuzz
    implicit none
    integer :: i = 1
     
    do i = 1, 100
        if(((Mod(i,3)==0)) .and. ((Mod(i,5) == 0)))then
        print *, "fizzbuzz"
        elseif((Mod(i,3)==0))then
            print *, "fizz"
        elseif((Mod(i,5)==0))then
            print *, "buzz"
        else
            print *, i
        endif
                
    end do
    end program FizzBuzz

File: Prolog/FizzBuzz.pl
:- use_module(library(aggregate),[foreach/2]).
:- initialization(fizzbuzz).


fizzbuzz :-
        foreach(between(1,100,A), print_fizzbuzz(A)).
 
print_fizzbuzz(A) :-
        (  0 is A mod 15
        -> print('fizzbuzz')
        ;  0 is A mod 3
        -> print('fizz')
        ;  0 is A mod 5
        -> print('buzz')
        ;  print(A)
        ),
        nl.